src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/rewriter/term/TermFieldRewriterTest.java

```diff
@@ -0,0 +1,124 @@
+
+    @Before
+    public void setup() throws IOException {
+        URL url = Resources.getResource(TEST_MAPPING_FILE);
+        String mappings = Resources.toString(url, Charsets.UTF_8);
+        LocalClusterState.state(null);
+        mockLocalClusterState(mappings);
+    }
+
+    @Test
+    public void testFromSubqueryShouldPass() {
+        String sql = "SELECT t.age as a FROM (SELECT age FROM semantics WHERE employer = 'david') t";
+        String expected = "SELECT t.age as a FROM (SELECT age FROM semantics WHERE employer.keyword = 'david') t";
+
+        assertThat(rewriteTerm(sql),
+                IsEqualIgnoreCaseAndWhiteSpace.equalToIgnoreCaseAndWhiteSpace(expected));
+    }
+
+    @Test
+    public void testFromSubqueryWithoutTermShouldPass() {
+        String sql = "SELECT t.age as a FROM (SELECT age FROM semantics WHERE age = 10) t";
+        String expected = sql;
+
+        assertThat(rewriteTerm(sql),
+                IsEqualIgnoreCaseAndWhiteSpace.equalToIgnoreCaseAndWhiteSpace(expected));
+    }
+
+    private String rewriteTerm(String sql) {
+        SQLQueryExpr sqlQueryExpr = SqlParserUtils.parse(sql);
+        sqlQueryExpr.accept(new TermFieldRewriter());
+        return SQLUtils.toMySqlString(sqlQueryExpr)
+                .replaceAll("[\\n\\t]+", " ")
+                .replaceAll("^\\(", " ")
+                .replaceAll("\\)$", " ")
+                .trim();
+    }
+
+    /**
+     * Tests if a string is equal to another string, ignore the case and whitespace.
+     */
+    public static class IsEqualIgnoreCaseAndWhiteSpace extends TypeSafeMatcher<String> {
+        private final String string;
+
+        public IsEqualIgnoreCaseAndWhiteSpace(String string) {
+            if (string == null) {
+                throw new IllegalArgumentException("Non-null value required");
+            }
+            this.string = string;
+        }
+
+        @Override
+        public boolean matchesSafely(String item) {
+            return ignoreCase(ignoreSpaces(string)).equals(ignoreCase(ignoreSpaces(item)));
+        }
+
+        @Override
+        public void describeMismatchSafely(String item, Description mismatchDescription) {
+            mismatchDescription.appendText("was ").appendValue(item);
+        }
+
+        @Override
+        public void describeTo(Description description) {
+            description.appendText("a string equal to ")
+                    .appendValue(string)
+                    .appendText(" ignore case and white space");
+        }
+
+        public String ignoreSpaces(String toBeStripped) {
+            return toBeStripped.replaceAll("\\s+", "").trim();
+        }
+
+        public String ignoreCase(String toBeLower) {
+            return toBeLower.toLowerCase();
+        }
+
+        public static Matcher<String> equalToIgnoreCaseAndWhiteSpace(String expectedString) {
+            return new IsEqualIgnoreCaseAndWhiteSpace(expectedString);
+        }
+    }
+}
```

Comment:
Would it be better to move it to utils/stringutils?

Reply:
Make sense. Done.